packages/perseus/src/widgets/grapher/util.tsx

```diff
@@ -1,661 +1,35 @@
-    // this function ensures that a, b > 0, and c is its
-    // smallest possible positive value.
-    let amplitude = coeffs[0];
-    let angularFrequency = coeffs[1];
-    let phase = coeffs[2];
-    const verticalOffset = coeffs[3];
-
-    // Guarantee a > 0
-    if (amplitude < 0) {
-        amplitude *= -1;
-        angularFrequency *= -1;
-        phase *= -1;
-    }
-
-    const period = Math.PI;
-    // Guarantee b > 0
-    if (angularFrequency < 0) {
-        angularFrequency *= -1;
-        phase *= -1;
-        phase += period / 2;
-    }
-
-    // Guarantee c is smallest possible positive value
-    while (phase > 0) {
-        phase -= period;
-    }
-    while (phase < 0) {
-        phase += period;
-    }
-
-    return [amplitude, angularFrequency, phase, verticalOffset];
-}
-
-const PlotDefaults = {
-    areEqual: function (
-        coeffs1: ReadonlyArray<number>,
-        coeffs2: ReadonlyArray<number>,
-    ): boolean {
-        return Util.deepEq(coeffs1, coeffs2);
-    },
-    Movable: Plot,
-    getPropsForCoeffs: function (coeffs: ReadonlyArray<number>): {fn: any} {
-        return {
-            // @ts-expect-error - TS2339 - Property 'getFunctionForCoeffs' does not exist on type '{ readonly areEqual: (coeffs1: any, coeffs2: any) => boolean; readonly Movable: any; readonly getPropsForCoeffs: (coeffs: any) => any; }'.
-            fn: _.partial(this.getFunctionForCoeffs, coeffs),
-        };
-    },
-} as const;
-
-const Linear: LinearType = _.extend({}, PlotDefaults, {
-    url: "https://ka-perseus-graphie.s3.amazonaws.com/67aaf581e6d9ef9038c10558a1f70ac21c11c9f8.png",
-
-    defaultCoords: [
-        [0.25, 0.75],
-        [0.75, 0.75],
-    ],
-
-    getCoefficients: function (
-        coords: Coords,
-    ): ReadonlyArray<number> | undefined {
-        const p1 = coords[0];
-        const p2 = coords[1];
-
-        const denom = p2[0] - p1[0];
-        const num = p2[1] - p1[1];
-
-        if (denom === 0) {
-            return;
-        }
-
-        const m = num / denom;
-        const b = p2[1] - m * p2[0];
-        return [m, b];
-    },
-
-    getFunctionForCoeffs: function (coeffs: ReadonlyArray<number>, x: number) {
-        const m = coeffs[0];
-        const b = coeffs[1];
-        return m * x + b;
-    },
-
-    getEquationString: function (coords: Coords) {
-        const coeffs: ReadonlyArray<number> = this.getCoefficients(coords);
-        const m: number = coeffs[0];
-        const b: number = coeffs[1];
-        return "y = " + m.toFixed(3) + "x + " + b.toFixed(3);
-    },
-});
-
-const Quadratic: QuadraticType = _.extend({}, PlotDefaults, {
-    url: "https://ka-perseus-graphie.s3.amazonaws.com/e23d36e6fc29ee37174e92c9daba2a66677128ab.png",
-
-    defaultCoords: [
-        [0.5, 0.5],
-        [0.75, 0.75],
-    ],
+type MovableMap = {
+    [K in keyof typeof GrapherUtil.MOVABLES]: any;
+};
+export const movableToGraphieMovable: MovableMap = {
```

Comment:
I was slightly confused by the name of this object. Could it be called something like `movableTypeToComponent` or `movableToComponentMap`? I guess the `movable` vs. `GraphieMovable` distinction confused me.